pandas/tests/frame/test_arithmetic.py

```diff
@@ -127,132 +126,86 @@ def test_df_add_flex_filled_mixed_dtypes(self):
-                result = getattr(frame, op)(2 * frame)
-                exp = f(frame, 2 * frame)
-                tm.assert_frame_equal(result, exp)
-
-                # vs mix float
-                result = getattr(mixed_float, op)(2 * mixed_float)
-                exp = f(mixed_float, 2 * mixed_float)
-                tm.assert_frame_equal(result, exp)
-                _check_mixed_float(result, dtype=dict(C=None))
-
-                # vs mix int
-                if op in ['add', 'sub', 'mul']:
-                    result = getattr(mixed_int, op)(2 + mixed_int)
-                    exp = f(mixed_int, 2 + mixed_int)
-
-                    # no overflow in the uint
-                    dtype = None
-                    if op in ['sub']:
-                        dtype = dict(B='uint64', C=None)
-                    elif op in ['add', 'mul']:
-                        dtype = dict(C=None)
-                    tm.assert_frame_equal(result, exp)
-                    _check_mixed_int(result, dtype=dtype)
-
-                    # rops
-                    r_f = lambda x, y: f(y, x)
-                    result = getattr(frame, 'r' + op)(2 * frame)
-                    exp = r_f(frame, 2 * frame)
-                    tm.assert_frame_equal(result, exp)
-
-                    # vs mix float
-                    result = getattr(mixed_float, op)(2 * mixed_float)
-                    exp = f(mixed_float, 2 * mixed_float)
-                    tm.assert_frame_equal(result, exp)
-                    _check_mixed_float(result, dtype=dict(C=None))
-
-                    result = getattr(intframe, op)(2 * intframe)
-                    exp = f(intframe, 2 * intframe)
-                    tm.assert_frame_equal(result, exp)
-
-                    # vs mix int
-                    if op in ['add', 'sub', 'mul']:
-                        result = getattr(mixed_int, op)(2 + mixed_int)
-                        exp = f(mixed_int, 2 + mixed_int)
-
-                        # no overflow in the uint
-                        dtype = None
-                        if op in ['sub']:
-                            dtype = dict(B='uint64', C=None)
-                        elif op in ['add', 'mul']:
-                            dtype = dict(C=None)
-                        tm.assert_frame_equal(result, exp)
-                        _check_mixed_int(result, dtype=dtype)
-            except:
-                printing.pprint_thing("Failing operation %r" % op)
-                raise
-
-            # ndim >= 3
-            ndim_5 = np.ones(frame.shape + (3, 4, 5))
+    def test_arith_flex_frame(self, all_arithmetic_operators, float_frame,
+                              mixed_float_frame):
+        # one instance of parametrized fixture
+        op = all_arithmetic_operators
+
+        def f(x, y):
+            if op.startswith('__r'):
+                # get op without "r" and invert it
+                return getattr(operator, op.replace('__r', '__'))(y, x)
```

Comment:
@jbrockmendel pls have a look here

Reply:
This is fine.  The alternative would be to do a lookup in core.ops, but I think for a reader who is unfamiliar with core.ops this is clearer.

Reply:
Modified comment slightly